### DISCUSSION TITLE
Is there a way to use 'eleventyExcludeFromCollections' in the eleventyComputed property?

### DISCUSSION THREAD

**Discussion by mark-reason** [11ty/eleventy#2211] [Uncategorized]
I'm trying to use page data to decide whether a page is excluded from collections, something like this:
```javascript
---js
{
  layout: "layouts/base.html",
  pagination: {
    data: "pages",
    alias: "pg",
    size: 1,
    addAllPagesToCollections: true,
  },
  eleventyComputed: {
    title: (data) => data.pg.title,
    eleventyExcludeFromCollections: (data) => {
      return data.pg.permalink === '/excluded-page';
    },
  }
}
---
```

However this seems to always return 'true' even if it's not. Even if I just 'return false' then all pages are then excluded from collections.

**Comment by pdehaan** [comment]
I would have expected your solution to work, but I'm struggling to figure out locally why it isnt working for me.
My quick workaround was to create a custom collection based on a tag and just re-filter the collection based on `eleventyExcludeFromCollections` frontmatter.

```js
// .eleventy.js snippet
    eleventyConfig.addCollection("pages", function (collectionApi) {
        const pages = [...collectionApi.getFilteredByTag("pages")];
        return pages.filter(p => {
            return !p.data.eleventyExcludeFromCollections;
        });
    });
```

```njk
---
# src/pagination.njk
pagination:
    data: "pages"
    alias: "pg"
    size: 1
    addAllPagesToCollections: true
permalink: "{{ pg.permalink }}"
tags:
    - pages
---

<h1>title={{ title }}</h1>
<p>permalink={{ pg.permalink }}</p>
<p>eleventyExcludeFromCollections={{ eleventyExcludeFromCollections }}</p>
<p>draft={{ pg.draft }}</p>
<p>{{ page | inspect }}</p>
```

```js
// src/pagination.11tydata.js (because I find it easier than JSON front matter. 🤷 )
module.exports = () => {
  return {
    eleventyComputed: {
      title(data) {
        return data.pg.title;
      },
      eleventyExcludeFromCollections(data) {
        if (data.pg.draft || data.page.url === "/excluded-page/") {
            return true;
        }
        return false;
      },
    },
  };
};
```

### src/_data/pages.json
```js
[
    {"title": "Page 1", "permalink": "/page-number-one/" },
    {"title": "Page two", "permalink": "/nested/page/two/", "draft": true},
    {"title": "secret page", "permalink": "/excluded-page/"}
]
```

```njk
---
# src/collection-test.njk
---
{% for p in collections.pages %}
<p>{{ p.data.title }} {{ p.url }} {{ p.data.draft | inspect }}</p>
{% endfor %}
```

### OUTPUT

```html
<p>Page 1 /page-number-one/ undefined</p>
```

Note that "Page 2" isn't listed (because we set custom `draft: true` flag) and "secret page" isn't listed (because it's generated url/permalink matches that disallow list).

**Reply by darekkay** [reply]
> I would have expected your solution to work, but I'm struggling to figure out locally why it isnt working for me.

I think this note on the [Computed Data page](https://www.11ty.dev/docs/data-computed/) provides the answer:

> It is important to note that Computed Data is computed right before templates are rendered. Therefore Computed Data cannot be used to modify the [special data properties used to configure templates](https://www.11ty.dev/docs/data-configuration/) (e.g. layout, pagination, tags etc.). These restrictions may be relaxed over time.
